sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -555,55 +555,56 @@ public TableServiceVersion getServiceVersion() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Void> deleteEntity(String partitionKey, String rowKey) {
-        return deleteEntityWithResponse(partitionKey, rowKey, null).flatMap(FluxUtil::toMono);
+        return deleteEntityWithResponse(partitionKey, rowKey, null, false, null).flatMap(FluxUtil::toMono);
     }
 
     /**
      * Deletes an entity from the table.
      *
-     * @param tableEntity The table entity to delete.
+     * @param entity The table entity to delete.
+     * @param ifUnchanged When true, the ETag of the provided entity must match the ETag of the entity in the Table
+     * service. If the values do not match, the update will not occur and an exception will be thrown.
      *
-     * @return An empty reactive result.
+     * @return A reactive result containing an HTTP response.
      *
-     * @throws IllegalArgumentException If the {@link TableEntity provided entity}'s partition key or row key are
-     * {@code null} or empty.
+     * @throws IllegalArgumentException If the provided partition key or row key are {@code null} or empty.
      * @throws TableServiceException If the request is rejected by the service.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Void> deleteEntity(TableEntity tableEntity) {
-        return deleteEntityWithResponse(tableEntity.getPartitionKey(), tableEntity.getRowKey(), tableEntity.getETag(),
-            null).flatMap(FluxUtil::toMono);
+    public Mono<Void> deleteEntity(TableEntity entity, boolean ifUnchanged) {
```

Comment:
This method can just take the entity as the next method takes `entity` and `ifUnchanged` params.

Reply:
Should we do the same for the sync client? This is, having the following:

```java
public void deleteEntity(String partitionKey, String rowKey)
public void deleteEntity(TableEntity tableEntity)
public Response<Void> deleteEntityWithResponse(TableEntity tableEntity, boolean ifUnchanged, Duration timeout, Context context)
```

While the async client would look have the following APIs:

```java
public Mono<Void> deleteEntity(String partitionKey, String rowKey)
public Mono<Void> deleteEntity(TableEntity tableEntity)
public Mono<Response<Void>> deleteEntityWithResponse(TableEntity tableEntity, boolean ifUnchanged)
```